crates/tabby/src/services/completion.rs

```diff
@@ -109,19 +109,55 @@ pub struct Segments {
     suffix: Option<String>,
 
     /// The relative path of the file that is being edited.
-    /// When git_url is set, this is the path of the file in the git repository.
-    /// When git_url is empty, this is the path of the file in the workspace.
+    /// - When `git_url` is set, this is the path of the file in the git repository.
+    /// - When `git_url` is empty, this is the path of the file in the workspace.
     filepath: Option<String>,
 
     /// The remote URL of the current git repository.
     /// Leave this empty if the file is not in a git repository,
     /// or the git repository does not have a remote URL.
     git_url: Option<String>,
 
+    /// The relevant code snippets provided by editor.
+    relevant_snippets: Option<Vec<RelevantSnippet>>,
+
```

Comment:
```suggestion
    /// The relevant code snippets provided by editor.
    declarations: Option<Vec<Declaration>>,
    
  struct Declaration {
    filepath: string
    body: string
  }
```

Please don't reuse type for different fields. Let's use simple declaration mechanism for now.